src/src/PluginStructs/P020_data_struct.cpp

```diff
@@ -196,35 +196,58 @@ void P020_Task::discardSerialIn() {
 // We can also use the rules engine for local control!
 void P020_Task::rulesEngine(String message) {
   if (!(Settings.UseRules)) { return; }
-  int NewLinePos = message.indexOf(F("\r\n"));
+  int NewLinePos    = 0;
+  uint16_t StartPos = 0;
 
-  if (NewLinePos > 0) { message = message.substring(0, NewLinePos); }
-  String eventString;
+  NewLinePos = message.indexOf(F("\r\n"), StartPos);
```

Comment:
You should scan for `'\n'` and just ignore any `'\r'`

Reply:
I've been thinking about that, I'll change it.

Reply:
Changed.